UM/Mesh/MeshData.py

```diff
@@ -405,6 +409,24 @@ def invertNormals(self) -> None:
                 new_vertices.append(self._vertices[i + 2])
             self._vertices = NumPyUtil.immutableNDArray(new_vertices)
 
+    def calculateUnwrappedUVCoordinates(self) -> Optional[tuple[int, int]]:
+        """Create a new set of unwrapped texture coordinates for the mesh."""
+        # remove_indices = False
+        if self._indices is None:
+            if self._vertices is None or self._vertex_count < 3:
+                return None
+            # remove_indices = True
+            indices = numpy.arange(self._vertex_count, dtype=numpy.int32).reshape(-1, 3)  # 3 verts per sub-array.
+        else:
+            indices = numpy.array()
+
+        self._uvs, texture_width, texture_height = uvula.unwrap(self._vertices, indices)
```

Comment:
The `unwrap` function can throw
(see;
https://github.com/Ultimaker/libUvula/blob/4eb68c51af2fcaa10d4d70b8698b8fd31c729d93/pyUvula/pyUvula.cpp#L20
and;
https://github.com/Ultimaker/libUvula/blob/4eb68c51af2fcaa10d4d70b8698b8fd31c729d93/pyUvula/pyUvula.cpp#L38
).

Handling that would already pull one thing off of QA's list (the defensive programming thing) 😉